src/libpmemfile/CMakeLists.txt

```diff
@@ -31,7 +31,9 @@
 
 set(SOURCES path_resolve.c preload.c)
 
-add_c_flag(-Wmissing-field-initializers)
+if("${CMAKE_C_COMPILER_ID}" MATCHES "Clang")
+	add_c_flag(-Wno-missing-field-initializers)
+endif()
```

Comment:
Oh, right, or just add the flag in the top level CMakeLists.txt, I don't think there is a need for a function for this.